### PR TITLE
Enclose The Robolectric.buildActivity() call with in try

### PR DESCRIPTION
**Pull request by arinmodi**
Overview
Fixes : https://github.com/robolectric/robolectric/issues/7439

Proposed Changes
There are some direct callings of Robolectric.buildActivity in ShadowActivityTest without try-with-resource grammer to close ActivityController after usage, and the Android Studio shows some warnings for us. It's better if we can use try-with-resource to auto-close ActivityController after test.

### CODE REVIEW

**Comment by arinmodi** [comment]
@utzcoz, can you please review this ?

**Comment by utzcoz** [comment]
Thanks @arinmodi . Could you squash commits to one?

**Comment by arinmodi** [comment]
@utzcoz, squashed the commits

File: robolectric/src/test/java/org/robolectric/shadows/ShadowActivityTest.java [hunk 1]
```diff
@@ -259,18 +263,21 @@ public void startActivityForResultAndReceiveResult_shouldSendResponsesBackToActi
...
+      activity.startActivityForResult(new Intent().setType("audio/*"), 123);
+      activity.startActivityForResult(new Intent().setType("image/*"), 456);
+
+      Intent requestIntent = new Intent().setType("video/*");
+      try {
+        shadowOf(activity)
+            .receiveResult(
+                requestIntent, Activity.RESULT_OK, new Intent().setData(Uri.parse("content:foo")));
+        fail();
+      } catch (Exception e) {
```
Comment (utzcoz) [inline robolectric/src/test/java/org/robolectric/shadows/ShadowActivityTest.java:278]: nit: we can move this `catch (Exception e)` to outer `try` statement, and remove this internal `try-catch`. The `try-with-resource` statement will auto call resource's `close` method at finally, see https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html.

Comment (utzcoz) [inline robolectric/src/test/java/org/robolectric/shadows/ShadowActivityTest.java:278]: nit: we can move this `catch (Exception e)` to outer `try` statement, and remove this internal `try-catch`. The `try-with-resource` statement will auto call resource's `close` method at finally, see https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html.

**Review by utzcoz — APPROVED** [review]

**Review by utzcoz — APPROVED** [review]
LGTM. Thanks.

**Comment by arinmodi** [comment]
@utzcoz, can you merge this ?
Thanks.

**Comment by utzcoz** [comment]
@arinmodi Please patience. I want @hoisie to give an extra reviewing for this PR. Thanks.

**Comment by utzcoz** [comment]
Thanks @arinmodi , merged.

**Pull request merged**